Chore(experiment): Adds AWS ECS experiment docs (#25)
chaosnative/developer-hub-cn: docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md

- Ensure to create a Kubernetes secret having the AWS access configuration(key) in the `CHAOS_NAMESPACE`. A sample secret file looks like:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: cloud-secret
type: Opaque
stringData:
  cloud_config.yml: |-
    # Add the cloud AWS credentials respectively
    [default]
    aws_access_key_id = XXXXXXXXXXXXXXXXXXX
    aws_secret_access_key = XXXXXXXXXXXXXXX
```

- If you change the secret key name (from `cloud_config.yml`) please also update the `AWS_SHARED_CREDENTIALS_FILE` ENV value on `experiment.yaml`with the same name.

## Default Validations

:::info

- ECS container instance should be in healthy state.

:::

## Experiment tunables

<details>
    <summary>Check the Experiment Tunables</summary>
    <h2>Mandatory Fields</h2>
    <table>
        <tr>
        <th> Variables </th>
        <th> Description </th>
        <th> Notes </th>
        </tr>
        <tr> 
        <td> CLUSTER_NAME </td>
        <td> Name of the target ECS cluster</td>

>         <td>  </td>

Reviewer: ```suggestion
        <td> Eg. cluster-1 </td>
```

        </tr>
        <tr>
        <td> REGION </td>
        <td> The region name of the target ECS cluster</td>

>         <td> </td>

Reviewer: ```suggestion
        <td> Eg. us-east-1 </td>
```

---
id: ecs-container-cpu-hog
title: ECS Container CPU Hog
---

## Introduction

> - ECS Container CPU hog contains chaos to disrupt the state of infra resources. The experiment can induce a stress chaos on AWS ECS container using Amazon SSM Run Command, this is carried out by using SSM Docs which is in-built in the experiment for the give chaos experiment.

Reviewer 2: ```suggestion
- ECS Container CPU hog contains chaos to disrupt the state of infra resources. The experiment can induce stress chaos on the AWS ECS container using Amazon SSM Run Command, this is carried out by using SSM Docs which is in-built into the experiment for the give chaos experiment.
```

- It causes CPU chaos on containers of ECS task with given `CLUSTER_NAME` using an SSM docs for a certain chaos duration.

- To select the Task Under Chaos (TUC) you can make use of servie name associated with the task that is, if you provide the service name along with cluster name only all the tasks associated with the given service will be selected as chaos targets.

- It tests the ECS task sanity (service availability) and recovery of the task containers subjected to CPU stress.

:::tip Fault execution flow chart
![ECS Container CPU Hog](./static/images/ecs-stress-chaos.png)
:::

## Uses

<details>
<summary>View the uses of the experiment</summary>
<div>

> Disk Pressure or CPU hogs is another very common and frequent scenario we find with conrainers/applications that can result in the eviction of the application (task container) and impact its delivery. Such scenarios that can still occur despite whatever availability aids docker provides. These problems are generally referred to as "Noisy Neighbour" problems.

Reviewer 2: ```suggestion
CPU hogs are another very common and frequent scenario we find with containers/applications that can result in the eviction of the application (task container) and impact its delivery. Such scenarios can still occur despite whatever availability aids docker provides. These problems are generally referred to as "Noisy Neighbour" problems.
```
